app/experimenter/experiments/api/v2/serializers.py

```diff
@@ -309,6 +307,7 @@ def validate(self, data):
         return data
 
     def update(self, instance, validated_data):
+        validated_data_copy = validated_data.copy()
```

Comment:
Since the pop is destructive, depending on the ordering of parent/child methods they could wind up with `validated_data` with or without `variants`.  Maybe it'd help if instead of popping from the passed in reference, you pop from the copy, so any parent/child invocations always get the unmodified total `variants_data`?